api/nakadi-event-bus-api.yaml

```diff
@@ -1648,6 +1698,25 @@ definitions:
         description: |
           Offset of the event being pointed to.
 
+  ShiftedCursor:
+    allOf:
+      - $ref: "#/definitions/Cursor"
+      - type: object
+        properties:
+          shift:
```

Comment:
it's probably worth documenting (or clarifying) the following - :

- what happens when shift is 0

- what happens when shift moves back beyond the stream horizon. Clients can't reason about this, they'll just have to make a call. If the server "rounds down" to the stream's horizon, that's possibly going to break someone being creative with using the shift to count on the consumer (we have real world experience people will count things if they're given anything that remotely looks like a number :) If the server rejects the request, the client's going to have to play a game of hi-lo.

- what happens when shift moves ahead of now, the head of the stream. Again, clients can't reason about this.

- when the number exceeds int32 plus/minus. If it's meant to be a long, that's worth saying. I've got no problem imagining a Nakadi stream with more than 2^32 events - volumes only ever go up :)

- if large numbers are allowed whether JSON libs can mess them up (eg some systems protect against this by drilling very large numbers through strings - as ugly as that is, it stops bugs caused by rounding)